### PR TITLE
Introduce DKMS support for DEB packages

### PR DESCRIPTION
**Pull request by robertbaldyga**
Signed-off-by: Robert Baldyga <robert.baldyga@intel.com>

### CODE REVIEW

**Review by rafalste — APPROVED** [review]

**Review by imjfckm — APPROVED** [review]

File: modules/Makefile [hunk 1]
```diff
@@ -16,8 +16,9 @@ obj-y += cas_disk/
 else
 
 OCFDIR=$(PWD)/../ocf
-KERNEL_DIR ?= "/lib/modules/$(shell uname -r)/build"
 PWD=$(shell pwd)
+KERNEL_VERSION ?= "$(shell uname -r)"
+KERNEL_DIR ?= "/lib/modules/$(KERNEL_VERSION)/build"
```
Comment (arutk) [inline modules/Makefile:21]: Looks like this breaks building CAS with kernel provided by source path (KERNEL_DIR=<kernel source> make ...). You can keep the old implementation and get the kernel version by "make -C $KERNEL_DIR kernelversion"

File: utils/pckgen.d/deb/debian/CAS_NAME-modules.dkms [hunk 2]
```diff
@@ -0,0 +1,11 @@
+PACKAGE_NAME="<CAS_NAME>-modules"
```
Comment (arutk) [inline utils/pckgen.d/deb/debian/CAS_NAME-modules.dkms:1]: How is <CAS_NAME> replaced with actual name? Will the file name be changed as well ("CAS_NAME-modules.dkms") ?

File: modules/Makefile (hunk 1 above)
Comment (robertbaldyga) [reply modules/Makefile:21]: It doesn't. I intentionally left `?=` for both to make it possible to override either `KERNEL_VERSION` or `KERNEL_DIR`.

File: utils/pckgen.d/deb/debian/CAS_NAME-modules.dkms (hunk 2 above)
Comment (robertbaldyga) [reply utils/pckgen.d/deb/debian/CAS_NAME-modules.dkms:1]: Yep. This is done by `utils/pckgen` for all files in this directory.

File: modules/Makefile (hunk 1 above)
Comment (robertbaldyga) [reply modules/Makefile:21]: Given that it's not used anywhere else it doesn't seem to be an issue.

Comment (arutk) [reply modules/Makefile:21]: It's confusing at least. If KERNEL_VERSION is not used anywhere then I don't see any point in this change

**Pull request merged**